lib/http/http.c

```diff
@@ -556,7 +557,12 @@ gotheaders(struct http_cookie * H, uint8_t * buf, size_t buflen)
 	if ((clen = http_findheader(H->res.headers, H->res.nheaders,
 	    "Content-Length")) != NULL) {
 		/* Parse the value (strtoull skips LWS). */
-		len = strtoull(clen, NULL, 0);
+		len_ull = strtoull(clen, NULL, 0);
+		if (len_ull > SIZE_MAX) {
```

Comment:
Perhaps we should also check for `ULLONG_MAX`, in case the value doesn't fit into an unsigned long long?

Reply:
I think you mean "check for `ERANGE`", but yes.

Reply:
oh wait, I misread the docs.  Yes, I agree.